src/Data/ExistsRE.purs

```diff
@@ -0,0 +1,20 @@
+module Data.ExistsRE where
+
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | A variant of `Exists` that works for type constructors that accept a _row_
+-- | of effects.
+data ExistsRE (f :: # ! -> *)
```

Comment:
I think Haskell does have the same problem, unless there's some type family kind of thing that can better represent it.

The `newtype` is definitely better, as that's how we're treating it, so I'll give it a go! It wouldn't matter whether we hid the constructor or not then too, as you wouldn't be able to do anything with it, it'd basically be the same as `exists eff. f eff` anyway.